### PR TITLE
bpo-41203: Replace OS X with macOS

### PR DESCRIPTION
**Pull request by pxeger**
I've made separate commits for changes in old whatsnew files because maybe they should be kept as is, for historical reasons?

Also I've committed changes from "Macintosh" to "Mac".

All of these changes were made with `sed` so I can't guarantee there are no false-positives, although I tried...

https://bugs.python.org/issue41203

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@pxeger

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/distutils/setupscript.rst [hunk 1]
```diff
@@ -675,7 +675,7 @@ information is sometimes used to indicate sub-releases.  These are
               'Intended Audience :: Developers',
               'Intended Audience :: System Administrators',
               'License :: OSI Approved :: Python Software Foundation License',
-              'Operating System :: MacOS :: MacOS X',
+              'Operating System :: MacOS :: MacmacOS',
```
Comment (ronaldoussoren) [inline Doc/distutils/setupscript.rst]: This one is incorrect for two reasons:

1. Spelling error in the new text
2. This value is a PyPI classifier, and those still use "MacOS X"

Please revert this change until "macOS" is added as an alternatieve to the PyPI classifiers.

File: Doc/glossary.rst [hunk 2]
```diff
@@ -1118,7 +1118,7 @@ Glossary
    universal newlines
       A manner of interpreting text streams in which all of the following are
       recognized as ending a line: the Unix end-of-line convention ``'\n'``,
-      the Windows convention ``'\r\n'``, and the old Macintosh convention
+      the Windows convention ``'\r\n'``, and the old Mac convention
```
Comment (ronaldoussoren) [inline Doc/glossary.rst]: I'd prefer to keep Macintosh here. This is referring to MacOS 9 and earlier, when Macintosh was commonly used to describe the system.

File: Doc/library/binascii.rst [hunk 3]
```diff
@@ -211,7 +211,7 @@ The :mod:`binascii` module defines the following functions:
       and 85.
 
    Module :mod:`binhex`
-      Support for the binhex format used on the Macintosh.
+      Support for the binhex format used on the Mac.
```
Comment (ronaldoussoren) [inline Doc/library/binascii.rst]: Same here, I'd prefer to keep Macintosh.

File: Doc/library/binhex.rst [hunk 4]
```diff
@@ -54,6 +54,6 @@ Notes
 There is an alternative, more powerful interface to the coder and decoder, see
 the source for details.
 
-If you code or decode textfiles on non-Macintosh platforms they will still use
-the old Macintosh newline convention (carriage-return as end of line).
+If you code or decode textfiles on non-Mac platforms they will still use
+the old Mac newline convention (carriage-return as end of line).
```
Comment (ronaldoussoren) [inline Doc/library/binhex.rst]: Likewise (for both lines).

File: Doc/library/codecs.rst [hunk 5]
```diff
@@ -1201,7 +1201,7 @@ particular, the following variants typically exist:
 | mac_latin2      | maclatin2, maccentraleurope,   | Central and Eastern Europe     |
 |                 | mac_centeuro                   |                                |
 +-----------------+--------------------------------+--------------------------------+
-| mac_roman       | macroman, macintosh            | Western Europe                 |
+| mac_roman       | macroman, Mac                  | Western Europe                 |
```
Comment (ronaldoussoren) [inline Doc/library/codecs.rst]: "macintosh" is the name of an encoding, don't change.

File: Doc/reference/lexical_analysis.rst [hunk 6]
```diff
@@ -50,7 +50,7 @@ A physical line is a sequence of characters terminated by an end-of-line
 sequence.  In source files and strings, any of the standard platform line
 termination sequences can be used - the Unix form using ASCII LF (linefeed),
 the Windows form using the ASCII sequence CR LF (return followed by linefeed),
-or the old Macintosh form using the ASCII CR (return) character.  All of these
+or the old Mac form using the ASCII CR (return) character.  All of these
```
Comment (ronaldoussoren) [inline Doc/reference/lexical_analysis.rst]: I'd prefer to keep the name "Macintosh" here as well.

File: Doc/whatsnew/2.0.rst [hunk 7]
```diff
@@ -574,7 +574,7 @@ mostly by Trent Mick of ActiveState.  (Confusingly, ``sys.platform`` is still
 treats code as 32 bit on Itanium.) PythonWin also supports Windows CE; see the
 Python CE page at http://pythonce.sourceforge.net/ for more information.
 
-Another new platform is Darwin/MacOS X; initial support for it is in Python 2.0.
+Another new platform is Darwin/MacmacOS; initial support for it is in Python 2.0.
```
Comment (ronaldoussoren) [inline Doc/whatsnew/2.0.rst]: Typo, this should be "macOS" instead of "MacmacOS".

File: Doc/whatsnew/2.0.rst [hunk 8]
```diff
@@ -773,7 +773,7 @@ the 'patches' and 'python-dev' lists at python.org.
 
 Recent versions of the GUSI development environment for MacOS support POSIX
 threads.  Therefore, Python's POSIX threading support now works on the
-Macintosh.  Threading support using the user-space GNU ``pth`` library was also
+Mac.  Threading support using the user-space GNU ``pth`` library was also
```
Comment (ronaldoussoren) [inline Doc/whatsnew/2.0.rst]: Don't change.

File: Doc/whatsnew/2.1.rst [hunk 9]
```diff
@@ -774,7 +774,7 @@ of the more notable changes are:
 * The size of the Unicode character database was shrunk by another 340K thanks
   to Fredrik Lundh.
 
-* Some new ports were contributed: MacOS X (by Steven Majewski), Cygwin (by
+* Some new ports were contributed: MacmacOS (by Steven Majewski), Cygwin (by
```
Comment (ronaldoussoren) [inline Doc/whatsnew/2.1.rst]: Typo.

File: Doc/whatsnew/2.2.rst [hunk 10]
```diff
@@ -1137,12 +1137,12 @@ bugs.  These figures are likely to be underestimates.
 Some of the more notable changes are:
 
 * The code for the MacOS port for Python, maintained by Jack Jansen, is now kept
-  in the main Python CVS tree, and many changes have been made to support MacOS X.
+  in the main Python CVS tree, and many changes have been made to support MacmacOS.
```
Comment (ronaldoussoren) [inline Doc/whatsnew/2.2.rst]: Typo.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
I'm generally in favour of this change, with some reservations:

1. I'd like to keep using "Macintosh" to refer to Mac systems before Darwin (Mac OS 9 and earlier)

2. There are changes to the various NEWS file, I'm not sure if this is a good idea. As far as I know we usually don't edit these.

There's also a number of incorrect changes (see detailed comments).

I haven't reviewed the change to the PDF file.

**Comment by pxeger** [comment]
RE changes to the NEWS files, I totally agree (especially the older ones) - it looks a bit anachronistic to have macOS references in the same breath as talking about Mac OS 9 and Macintosh. That's why I did them in separate commits, so it would be easier to cherrypick/rebase/whatever if they weren't wanted. Should I completely reset them?

File: Doc/library/binhex.rst [hunk 11]
```diff
@@ -11,7 +11,7 @@
 --------------
 
 This module encodes and decodes files in binhex4 format, a format allowing
-representation of Macintosh files in ASCII. Only the data fork is handled.
+representation of Mac files in ASCII. Only the data fork is handled.
```
Comment (serhiy-storchaka) [inline Doc/library/binhex.rst]: Same as above.

File: Doc/library/ctypes.rst [hunk 12]
```diff
@@ -1420,7 +1420,7 @@ correct library and dependencies are loaded.
 .. data:: DEFAULT_MODE
    :noindex:
 
-   The default mode which is used to load shared libraries.  On OSX 10.3, this is
+   The default mode which is used to load shared libraries.  On macOS 10.3, this is
```
Comment (serhiy-storchaka) [inline Doc/library/ctypes.rst]: It was Mac OS X for versions up to 10.7 inclusively.

File: Doc/library/os.rst [hunk 13]
```diff
@@ -301,7 +301,7 @@ process and user.
 
    .. note::
 
-      On Mac OS X, :func:`getgroups` behavior differs somewhat from
+      On macOS, :func:`getgroups` behavior differs somewhat from
```
Comment (serhiy-storchaka) [inline Doc/library/os.rst]: I have doubts about this one. If the note refers to 10.5 and earlier, it should be Mac OS X.

File: Doc/library/os.rst [hunk 14]
```diff
@@ -486,7 +486,7 @@ process and user.
 
    .. availability:: Unix.
 
-   .. note:: On Mac OS X, the length of *groups* may not exceed the
+   .. note:: On macOS, the length of *groups* may not exceed the
```
Comment (serhiy-storchaka) [inline Doc/library/os.rst]: Same as above.

**Comment by pxeger** [comment]
I have made the requested changes; please review again

File: Doc/library/shutil.rst [hunk 15]
```diff
@@ -26,7 +26,7 @@ copying and removal. For operations on individual files, see also the
    :func:`shutil.copy2`) cannot copy all file metadata.
 
    On POSIX platforms, this means that file owner and group are lost as well
-   as ACLs.  On Mac OS, the resource fork and other metadata are not used.
+   as ACLs.  On macOS, the resource fork and other metadata are not used.
```
Comment (serhiy-storchaka) [inline Doc/library/shutil.rst]: Is it relevant to modern macOS or only to classic MacOS?

File: Doc/library/site.rst [hunk 16]
```diff
@@ -177,7 +177,7 @@ Module contents
    Path to the user site-packages for the running Python.  Can be ``None`` if
    :func:`getusersitepackages` hasn't been called yet.  Default value is
    :file:`~/.local/lib/python{X.Y}/site-packages` for UNIX and non-framework Mac
```
Comment (serhiy-storchaka) [inline Doc/library/site.rst]: Fix Mac macOS.

File: Doc/library/site.rst [hunk 17]
```diff
@@ -187,7 +187,7 @@ Module contents
 
    Path to the base directory for the user site-packages.  Can be ``None`` if
    :func:`getuserbase` hasn't been called yet.  Default value is
-   :file:`~/.local` for UNIX and Mac OS X non-framework builds,
+   :file:`~/.local` for UNIX and macOS non-framework builds,
    :file:`~/Library/Python/{X.Y}` for Mac framework builds, and
```
Comment (serhiy-storchaka) [inline Doc/library/site.rst:191]: What is the correct name?

File: Doc/library/tkinter.ttk.rst [hunk 18]
```diff
@@ -759,7 +759,7 @@ ones inherited from :class:`ttk.Widget`.
 Platform-specific notes
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-* On MacOS X, toplevel windows automatically include a built-in size grip
+* On MacmacOS, toplevel windows automatically include a built-in size grip
```
Comment (serhiy-storchaka) [inline Doc/library/tkinter.ttk.rst]: Fix MacmacOS.

File: Doc/library/urllib.request.rst [hunk 19]
```diff
@@ -164,8 +164,8 @@ The :mod:`urllib.request` module defines the following functions:
    This helper function returns a dictionary of scheme to proxy server URL
    mappings. It scans the environment for variables named ``<scheme>_proxy``,
    in a case insensitive approach, for all operating systems first, and when it
-   cannot find it, looks for proxy information from Mac OSX System
-   Configuration for Mac OS X and Windows Systems Registry for Windows.
+   cannot find it, looks for proxy information from macOS System
```
Comment (serhiy-storchaka) [inline Doc/library/urllib.request.rst:167]: Is it the official name of the program?

File: Doc/library/urllib.request.rst [hunk 20]
```diff
@@ -298,8 +298,8 @@ The following classes are provided:
    the list of proxies from the environment variables
    ``<protocol>_proxy``.  If no proxy environment variables are set, then
    in a Windows environment proxy settings are obtained from the registry's
-   Internet Settings section, and in a Mac OS X environment proxy information
-   is retrieved from the OS X System Configuration Framework.
+   Internet Settings section, and in a macOS environment proxy information
+   is retrieved from the macOS System Configuration Framework.
```
Comment (serhiy-storchaka) [inline Doc/library/urllib.request.rst:302]: Same as above.

File: Doc/using/mac.rst [hunk 21]
```diff
@@ -2,13 +2,13 @@
 .. _using-on-mac:
 
 ***************************
-Using Python on a Macintosh
+Using Python on a Mac
```
Comment (serhiy-storchaka) [inline Doc/using/mac.rst:5]: Fix lines before and after this line.

Is Mac an official name of all computers running macOS?

File: Doc/using/mac.rst [hunk 22]
```diff
@@ -17,7 +17,7 @@ the IDE and the Package Manager that are worth pointing out.
 Getting and Installing MacPython
 ================================
 
-Mac OS X 10.8 comes with Python 2.7 pre-installed by Apple.  If you wish, you
+macOS 10.8 comes with Python 2.7 pre-installed by Apple.  If you wish, you
```
Comment (serhiy-storchaka) [inline Doc/using/mac.rst:20]: It was Mac OS X.

File: Doc/using/mac.rst [hunk 23]
```diff
@@ -148,10 +148,10 @@ X by Apple, and the latest version can be downloaded and installed from
 https://www.activestate.com; it can also be built from source.
 
 *wxPython* is another popular cross-platform GUI toolkit that runs natively on
-Mac OS X. Packages and documentation are available from https://www.wxpython.org.
+macOS. Packages and documentation are available from https://www.wxpython.org.
 
 *PyQt* is another popular cross-platform GUI toolkit that runs natively on Mac
```
Comment (serhiy-storchaka) [inline Doc/using/mac.rst:153]: Fix Mac macOS.

File: Doc/whatsnew/2.2.rst [hunk 24]
```diff
@@ -1137,12 +1137,12 @@ bugs.  These figures are likely to be underestimates.
...
 
   The most significant change is the ability to build Python as a framework,
   enabled by supplying the :option:`!--enable-framework` option to the configure
   script when compiling Python.  According to Jack Jansen, "This installs a
-  self-contained Python installation plus the OS X framework "glue" into
+  self-contained Python installation plus the macOS framework "glue" into
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/2.2.rst]: Do not change a cite.

File: Doc/whatsnew/2.2.rst [hunk 25]
```diff
@@ -1151,14 +1151,14 @@ Some of the more notable changes are:
...
-  windowing, QuickTime, scripting, etc. have been ported to OS X, but they've been
+  windowing, QuickTime, scripting, etc. have been ported to macOS, but they've been
   left commented out in :file:`setup.py`.  People who want to experiment with
   these modules can uncomment them manually.
 
   .. Jack's original comments:
      The main change is the possibility to build Python as a
      framework. This installs a self-contained Python installation plus the
-     OSX framework "glue" into /Library/Frameworks/Python.framework (or
+     macOS framework "glue" into /Library/Frameworks/Python.framework (or
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/2.2.rst]: Same as above.

File: Doc/whatsnew/2.3.rst [hunk 26]
```diff
@@ -375,7 +375,7 @@ PEP 278: Universal Newline Support
 ==================================
 
 The three major operating systems used today are Microsoft Windows, Apple's
-Macintosh OS, and the various Unix derivatives.  A minor irritation of
+Mac OS, and the various Unix derivatives.  A minor irritation of
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/2.3.rst]: Do not change.

**Review by serhiy-storchaka — COMMENTED** [review]
Do not use anachronical macOS for versions which predated the name macOS.

File: Doc/using/mac.rst (hunk 22 above)
Comment (pxeger) [reply Doc/using/mac.rst:20]: Current versions of macOS still have Python 2.7 I believe

File: Doc/using/mac.rst (hunk 23 above)
Comment (pxeger) [reply Doc/using/mac.rst:153]: ??

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
There are incorrect changes. My comments have not yet been addressed.

File: Doc/library/site.rst (hunk 17 above)
Comment (pxeger) [reply Doc/library/site.rst:191]: I think this still applies to current macOS

File: Doc/library/urllib.request.rst (hunk 19 above)
Comment (pxeger) [reply Doc/library/urllib.request.rst:167]: Don't think so

File: Doc/using/mac.rst (hunk 21 above)
Comment (pxeger) [reply Doc/using/mac.rst:5]: Yes?

**Comment by pxeger** [comment]
I have made the requested changes; please review again

File: Doc/library/platform.rst [hunk 27]
```diff
@@ -232,13 +232,13 @@ Windows Platform
    .. versionadded:: 3.8
 
 
-Mac OS Platform
+macOS Platform
 ---------------
```
Comment (serhiy-storchaka) [inline Doc/library/platform.rst:236]: The length of the underlying line should be corrected.

File: Doc/library/site.rst [hunk 28]
```diff
@@ -32,7 +32,7 @@ It starts by constructing up to four directories from a head and a tail part.
 For the head part, it uses ``sys.prefix`` and ``sys.exec_prefix``; empty heads
 are skipped.  For the tail part, it uses the empty string and then
 :file:`lib/site-packages` (on Windows) or
-:file:`lib/python{X.Y}/site-packages` (on Unix and Macintosh).  For each
+:file:`lib/python{X.Y}/site-packages` (on Unix and Mac).  For each
```
Comment (serhiy-storchaka) [inline Doc/library/site.rst:35]: macOS

File: Doc/library/site.rst (hunk 17 above)
Comment (serhiy-storchaka) [reply Doc/library/site.rst:191]: macOS?

File: Doc/library/urllib.request.rst (hunk 19 above)
Comment (serhiy-storchaka) [reply Doc/library/urllib.request.rst:167]: On Apple's site it is called just "System Configuration".

File: Doc/library/urllib.request.rst (hunk 20 above)
Comment (serhiy-storchaka) [reply Doc/library/urllib.request.rst:302]: I think it is just "the System Configuration framework".

File: Doc/using/mac.rst (hunk 21 above)
Comment (serhiy-storchaka) [reply Doc/using/mac.rst:5]: The number of `*`s should match the length of the title.

Comment (serhiy-storchaka) [reply Doc/using/mac.rst:5]: According to Wikipedia "Mac"  is a correct name for modern Apple's computers.

File: Doc/using/mac.rst (hunk 23 above)
Comment (serhiy-storchaka) [reply Doc/using/mac.rst:153]: "Mac macOS" -> "macOS"
